### PR TITLE
Allow updating local assets

### PR DESCRIPTION
**Pull request by MrFlashAccount**
### Pull Request Description

Closes: enso-org/cloud-v2#11166

This PR fixes a long-standing bug, that didn't allow to update assets on the local backend.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [ ] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [ ] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [ ] Unit tests have been written where possible.

### CODE REVIEW

**Review by somebody1234 — CHANGES_REQUESTED** [review]
we still want to be able to choose between renaming and overwriting when uploading to the local backend

File: app/gui/src/dashboard/hooks/backendHooks.ts [hunk 1]
```diff
@@ -149,6 +149,7 @@ const INVALIDATION_MAP: Partial<
   updateAsset: ['listDirectory', 'listAssetVersions'],
   closeProject: ['listDirectory', 'listAssetVersions'],
   updateDirectory: ['listDirectory'],
+  getProjectDetails: ['listDirectory'],
```
Comment (somebody1234) [inline app/gui/src/dashboard/hooks/backendHooks.ts]: is this correct? (why does a query need to invalidate another query?)

File: app/gui/src/dashboard/layouts/AssetsTable.tsx [hunk 2]
```diff
@@ -1896,7 +1896,7 @@ export default function AssetsTable(props: AssetsTableProps) {
 
                   fileMap.set(asset.id, conflict.file)
 
-                  insertAssets([asset], event.parentId)
+                  // insertAssets([asset], event.parentId)
```
Comment (somebody1234) [inline app/gui/src/dashboard/layouts/AssetsTable.tsx]: commented code

File: app/gui/src/dashboard/hooks/backendHooks.ts (hunk 1 above)
Comment (MrFlashAccount) [reply app/gui/src/dashboard/hooks/backendHooks.ts]: @somebody1234 
CleanShot 2024-10-14 at 11 43 53@2x
getProjectDetails is a mutation

Comment (somebody1234) [reply app/gui/src/dashboard/hooks/backendHooks.ts]: technically yes, but logically it is a query, it's only a mutation because i was being silly

Comment (somebody1234) [reply app/gui/src/dashboard/hooks/backendHooks.ts]: see https://github.com/enso-org/enso/pull/11268/commits/c8200f652920c4efa181c6c8e5a9e316ee7c1018

Comment (somebody1234) [reply app/gui/src/dashboard/hooks/backendHooks.ts]: well, specifically, either way it's not supposed to *change* the backend's state - so logically it shouldn't need to invalidate anything

**Comment by MrFlashAccount** [comment]
@somebody1234  ready for re-CR I think

File: app/ide-desktop/client/src/index.ts [hunk 3]
```diff
@@ -1,9 +1,11 @@
-/** @file Definition of an Electron application, which entails the creation of a rudimentary HTTP
+/**
```
Comment (MrFlashAccount) [inline app/ide-desktop/client/src/index.ts:1]: these changes are made by Prettier (on save)

**Review by somebody1234 — APPROVED** [review]

File: app/gui/src/dashboard/hooks/backendHooks.ts (hunk 1 above)
Comment (MrFlashAccount) [reply app/gui/src/dashboard/hooks/backendHooks.ts]: Agree

File: .prettierignore [hunk 4]
```diff
@@ -40,6 +40,7 @@ app/ide-desktop/lib/dashboard/dist/
 app/gui/view/documentation/assets/stylesheet.css
 app/rust-ffi/pkg
 app/gui/src/project-view/assets/font-*.css
+app/gui/mockDist
```
Comment (farmaazon) [inline .prettierignore]: Fixed differently in https://github.com/enso-org/enso/pull/11317/commits/e0310e57f7cc62bb52b8d124f78d8802fd53ff30

I think it is closer to built artifacts than generated files

**Review by PabloBuchu — APPROVED** [review]

**Pull request merged**